src/MockHttpClient/HttpRequestMessageExtensions.cs

```diff
@@ -130,7 +132,104 @@ public static bool HasHeader(this HttpRequestMessage request, string headerName,
 
             return false;
         }
-    }
 
+        /// <summary>
+        /// Determines whether the request contains the specified content as string.
+        /// </summary>
+        /// <param name="request">the request</param>
+        /// <param name="expectedContent">the expected content</param>
+        /// <returns></returns>
+        public static bool HasStringContent(this HttpRequestMessage request, string expectedContent)
+        {
+            if (request.Content == null)
+            {
+                throw new ArgumentException($"string content '{expectedContent}' was expected but content is null");
+            }
 
+            var content = request.Content.ReadAsStringAsync().Result;
+
+            return content.Equals(expectedContent);
+        }
+
+        /// <summary>
+        /// Determines whether the request contains the specified content as json object.
+        /// </summary>
+        /// <typeparam name="T">The type of the expected content</typeparam>
+        /// <param name="request">the request</param>
+        /// <param name="expectedContent">the expected content</param>
+        /// <returns></returns>
+        public static bool HasJsonContent<T>(this HttpRequestMessage request, T expectedContent)
```

Comment:
Could this do the argument null checks then call `HasJsonContent(request, x=>x.equals(expectedContext))`

Reply:
Good idea, but can you help me with the syntax please? I cannot get past the `ambiguous call` error. Otherwise I would just extract the common part into a separate method.